### PR TITLE
Add design documentation page

### PR DESCRIPTION
**Pull request by gctucker**
Rename the "overview" page to "design" and update it to match the current state of the API & Pipeline.

### CODE REVIEW

File: doc/design.md [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+[Kubernetes](https://kubernetes.io/) cluster or a
+[LAVA](https://www.lavasoftware.org/) lab.  Additional Runtime implementations
+could be on-demand [VMs in GCE](https://cloud.google.com/compute),
+[LabGrid](https://www.pengutronix.de/en/software/labgrid.html) and maybe SSH.
+It's of course possible to create custom ones for particular use-cases.
+
+### Email Report
+
+The current implementation for email reports is very minimalist.  This pipeline
+steps just listens for events about kernel revisions being done, then sends an
```
Comment (JenySadadia) [inline doc/design.md:145]:
```suggestion
step just listens for events about kernel revisions being done, then sends an
```

File: doc/design.md [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+The Scheduler step listens for a variety of pub/sub events and then looks for
+matches in the `scheduler` [YAML
+configuration](https://github.com/kernelci/kernelci-pipeline/blob/main/config/pipeline.yaml#L144).
+Every time an event matches an entry in the config, it will send a new node
+with the "running" state to the API and submit a job in the designated runtime
+environment.
+
+> **Note**: It's then up to the job itself to send the results to the API and
+> any files to storage.  The scheduler doesn't track the status of submitted
+> jobs, however the timeout steps will update the status of nodes that reach
```
Comment (JenySadadia) [inline doc/design.md:124]:
```suggestion
> jobs, however, the timeout step will update the status of nodes that reach
```

File: doc/design.md [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+* Pub/Sub mechanism via the API, with [Redis](https://redis.io/) to manage
+  message queues.  This can now be used to coordinate client-side functions and
+  recreate a full modular pipeline with no additional framework
+* [CloudEvent](https://cloudevents.io/) for the formatting of Pub/Sub events
+* is written for [Python 3.10](https://www.python.org/downloads/release/python-3100/)
+* relies on [JWT](https://jwt.io/) and [OAuth 2.0](https://oauth.net/2/)
+  authentication for inter-operability with other services
+* treats storage entirely separately from the API which purely handles data.
+  Several storage solutions can be supported, such as SSH for self-contained
+  systems, AzureFiles and S3 for Cloud deployments etc.
```
Comment (JenySadadia) [inline doc/design.md:66]:
```suggestion
  systems, AzureFiles, and S3 for Cloud deployments etc.
```

File: doc/design.md [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+state.
+
+### Tarball
+
+The Tarball step listens for pub/sub events about new revisions or `checkout`
+nodes.  When one is received, typically because the trigger pushed a new one to
+the API, it then updates a local git checkout of the full kernel source tree.
+Then it makes a tarball with the source code and pushes it to the storage.
+Finally, it sends an update to the `checkout` node to the API with the URL of
+the tarball added to the list of artifects and the status set to "available".
```
Comment (JenySadadia) [inline doc/design.md:111]:
```suggestion
the tarball added to the list of artifacts and the status set to "available".
```

File: doc/design.md (hunk 2 above)
Comment (gctucker) [reply doc/design.md:124]: No this is plural on purpose, there are several timeout services.  I guess I should rephrase it to make it clearer though.

File: doc/design.md (hunk 3 above)
Comment (gctucker) [reply doc/design.md:66]: Well this one is a matter of style, see [Oxford comma](https://en.wikipedia.org/wiki/Serial_comma).  But yes I would rather drop it in this case.

**Comment by gctucker** [comment]
Thanks for the review.

I'm making another PR to add a page about Early Access, then once that's merged I think we can review the ToC and weights etc.
